### PR TITLE
WIP :Fix nonconformityreport

### PR DESCRIPTION
**Pull request by Agaba-derrick**
requesting for reviews here 
@tendomart  @mozzy11 @rahul6603

### CODE REVIEW

File: frontend/src/components/Reports/Routine.js [hunk 1]
```diff
@@ -100,9 +100,8 @@ export const RoutineReportsMenu = {
         },
         {
           link:
-            config.serverBaseUrl +
-            "/Report?type=patient&report=haitiNonConformityBySectionReason",
-          label: <FormattedMessage id="sideNav.label.noncomformityreportsbyunit"/>,
+            config.serverBaseUrl +"/NonconformityReportsByUnit",
```
Comment (manishjha-04) [inline frontend/src/components/Reports/Routine.js:103]: Hey @Agaba-derrick  Over here add `link:"/RoutineReport?type=patient&report=haitiNonConformityBySectionReason`

File: frontend/src/App.js [hunk 2]
```diff
@@ -210,6 +210,13 @@ export default function App() {
                   component={() => <Home />}
                   role=""
                 />
+                <SecureRoute
+                path="/NonconformityReportsByUnit"
+                exact
+                component={() => NonconformityReportsByUnit}
+                role="Reports" // Define the appropriate role
+              />
+
```
Comment (manishjha-04) [inline frontend/src/App.js:219]: Hey @Agaba-derrick  remove this route its of no use      `path="/RoutineReport"`  would be handling it !
 instead in 
`frontend/src/components/Reports/routine/Index.js` add ` {type === "patient" &&
              report === "haitiNonConformityBySectionReason" &&
              (< NonconformityReportsByUnit/>)}` cwhich is the comparison of params

Comment (manishjha-04) [reply frontend/src/App.js:219]: instead in
`frontend/src/components/Reports/routine/Index.js` add  `{type === "patient" && report === "haitiNonConformityBySectionReason" && (< NonconformityReportsByUnit/>)}` which is the comparison of params

**Review by tendomart — COMMENTED** [review]
@Agaba-derrick  you can also show us some screenshots

File: frontend/src/components/Reports/routine/Nonconformity-Unit.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      }
+      const printableData = await response.json();
+      // Handle the printable data, e.g., open in a new window or initiate download
+    } catch (error) {
+      console.error("Error generating printable data:", error);
+    }
+  };
+
+  return (
+    <div>
```
Comment (tendomart) [inline frontend/src/components/Reports/routine/Nonconformity-Unit.js:73]: And what will happen is you happen to have another div here ?

do you think you may need 
```
return(
  <>
      //your code here
  </>
);
```

File: frontend/src/components/Reports/routine/Nonconformity-Unit.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  TableRow,
+  DatePicker,
+  Button, // Added Button component
+} from "@carbon/react";
+
+const NonconformityReportsByUnit = () => {
+  const [reports, setReports] = useState([]);
+  const [startDate, setStartDate] = useState(null);
+  const [endDate, setEndDate] = useState(null);
+  const [printableDataUrl, setPrintableDataUrl] = useState(null); // Added state for printable data URL
```
Comment (sherrif10) [inline frontend/src/components/Reports/routine/Nonconformity-Unit.js:19]: Remove unneccessary comments, otherwise its looks good.

**Comment by mozzy11** [comment]
Also properly format the code

**Comment by Agaba-derrick** [comment]
Screenshot from 2024-03-18 11-53-22
New component for Non conformity by reason

**Comment by mozzy11** [comment]
@Agaba-derrick any progress on this . 
can you also use these existing re-usable componenets here https://github.com/I-TECH-UW/OpenELIS-Global-2/tree/develop_3x/frontend/src/components/Reports/study/common ??

**Pull request closed without merging**